crates/neon/src/sys/tag.rs

```diff
@@ -137,3 +139,37 @@ pub unsafe fn check_object_type_tag(env: Env, object: Local, tag: &super::TypeTa
 pub unsafe fn is_bigint(env: Env, val: Local) -> bool {
     is_type(env, val, napi::ValueType::BigInt)
 }
+
+pub unsafe fn is_map(env: Env, val: Local) -> bool {
+    let global_object = unsafe {
+        let mut out: super::raw::Local = std::mem::zeroed();
+        super::scope::get_global(env, &mut out);
+        out
+    };
+
+    let map_literal = {
+        let mut out: super::raw::Local = std::mem::zeroed();
+        let literal = b"Map";
+        assert_eq!(
+            super::string::new(&mut out, env, literal.as_ptr(), literal.len() as _),
+            true
+        );
+        out
+    };
+
+    let map_constructor = {
```

Comment:
The steps to get this global can be somewhat expensive. We can make wrappers that cache the pointer in a `LocalKeyRoot<_>>`.

Take a look at the `Json` example for how it caches `JSON.stringify` and `JSON.parse`.

https://github.com/neon-bindings/neon/blob/1efcd4ea0890dfe55a5d4cee9661aaf232539406/crates/neon/src/types_impl/extract/json.rs#L46-L53